src/System/include/BipedalLocomotion/System/NamedTuple.h

```diff
@@ -0,0 +1,391 @@
+{
+
+    /** Underline tuple that can be generate from this named_tuple */
+    using UnderlineTuple = std::tuple<typename Params::Type...>;
+
+    /**
+     * Constructor
+     */
+    constexpr named_tuple()
+        : std::tuple<Params...>()
+    {
+    }
+
+    /**
+     * Constructor
+     */
+    named_tuple(Params&&... args)
+        : std::tuple<Params...>(std::forward<Params>(args)...)
+    {
+    }
+
+    /**
+     * Construct from a tuple.
+     * @param t a tuple having the same number of elements of of named_tuple.
+     * @note This constructor requires the type of the elements of the tuple to be convertible into
+     * Params.
+     */
+    template <typename... Args>
+    named_tuple(const std::tuple<Args...>& t)
+        : std::tuple<Params...>(t)
+    {
+        static_assert(sizeof...(Params) == sizeof...(Args));
+    }
+
+    /**
+     * Construct from a tuple.
+     * @param t a tuple having the same number of elements of of named_tuple.
+     * @note This constructor requires the type of the elements of the tuple to be convertible into
+     * Params.
+     */
+    template <typename... Args>
+    named_tuple(std::tuple<Args...>&& t)
+        : std::tuple<Params...>(std::forward<std::tuple<Args...>>(t))
+    {
+        static_assert(sizeof...(Params) == sizeof...(Args));
+    }
+
+    /**
+     * Copy from a tuple.
+     *
+     */
+    template <typename... Args>
+    named_tuple& operator=(const std::tuple<Args...>& t)
+    {
+        static_assert(sizeof...(Params) == sizeof...(Args));
+        this->copy_from_tuple(t);
+        return *this;
+    }
+
+    /**
+     * Extracts the Ith element from the named_tuple. I must be an integer value in `[0,
+     * sizeof...(Types))`.
+     * @param t the named_tuple
+     * @return A reference to the selected element of t.
+     */
+    template <std::size_t I>
+    const typename std::tuple_element<I, UnderlineTuple>::type&
+    get() const noexcept
+    {
+        const auto& param = (std::get<I>(static_cast<const std::tuple<Params...>&>(*this)));
+        return param.param;
+    }
+
+    /**
+     * Extracts the Ith element from the named_tuple. I must be an integer value in `[0,
+     * sizeof...(Types))`.
+     * @param t the named_tuple
+     * @return A reference to the selected element of t.
+     */
+    template <std::size_t I>
+    typename std::tuple_element<I, UnderlineTuple>::type& get() noexcept
+    {
+        auto& param = (std::get<I>(static_cast<std::tuple<Params...>&>(*this)));
+        return param.param;
+    }
+
+    /**
+     * Extracts the  element from the named_tuple associated to the hash `Hash`. `Hash` must be
+     * among one of the hash values associated to the tuple when created.
+     * @@return A reference to the selected element of t.
+     */
+    template <hash_type Hash> const auto& get_from_hash() const noexcept
+    {
+        constexpr std::size_t index = get_element_index<0, Hash>();
```

Comment:
The initial zero is a bit confusing. You could remove it, if you invert the template parameters of ``get_element_index``. I.e., you would have the following:
```c++
template <hash_type Hash, std::size_t I = 0>
    constexpr typename std::enable_if<(I < sizeof...(Params)),
                                      const std::size_t>::type static get_element_index() noexcept
```

In this way you can exploit the default value for ``I``

Reply:
Done! Now the code is more readable :)